pkg/speaker/layer2/arp_announer.go

```diff
@@ -37,7 +37,16 @@ type arpSpeaker struct {
 	ip2mac map[string]net.HardwareAddr
 }
 
-func (a *arpSpeaker) getMac(ip string) *net.HardwareAddr {
+func (a *arpAnnouncer) ContainsIP(ip string) bool {
+	for _, addr := range a.addrs {
+		if addr.Contains(net.ParseIP(ip)) {
+			return true
+		}
+	}
+	return false
+}
+
+func (a *arpAnnouncer) getMac(ip string) *net.HardwareAddr {
 	a.lock.Lock()
```

Comment:
Perhaps this `lock` can be replaced with a read-write lock.

Reply:
done.